src/components/PostDetails.tsx

```diff
@@ -1,106 +1,124 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+import { createComment, deleteComment, getPostComments } from './api/users';
+import { Comment } from '../types/Comment';
+
+type Props = {
+  selectedPost: Post | null;
+};
+
+export const PostDetails: React.FC<Props> = ({ selectedPost }) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isErrorComments, setIsErrorComments] = useState(false);
+  const [isShownForm, setIsShownForm] = useState(false);
+
+  const isNoComments =
+    selectedPost && comments.length === 0 && !isLoading && !isErrorComments;
+
+  useEffect(() => {
+    setIsErrorComments(false);
+    setIsShownForm(false);
+    if (selectedPost) {
+      setIsLoading(true);
+      getPostComments(selectedPost.id)
+        .then(setComments)
+        .catch(() => setIsErrorComments(true))
+        .finally(() => setIsLoading(false));
+    }
+  }, [selectedPost]);
+
+  const addComment = ({ name, email, body, postId }: Omit<Comment, 'id'>) => {
```

Comment:
The method name `addComment` is not following the recommended naming conventions for event handler functions in React. According to the conventions, the method should be named `handleAddComment`. It's a good practice to follow these conventions as it makes the code more readable and maintainable. Check this resource for more details: https://medium.com/javascript-in-plain-english/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364